src/internal_modules/roc_sndio/target_sox/roc_sndio/sox_sink.cpp

```diff
@@ -217,6 +217,14 @@ status::StatusCode SoxSink::write(audio::Frame& frame) {
     return status::StatusOK;
 }
 
+status::StatusCode SoxSink::flush() {
+    if (fflush((FILE*)output_->fp) == 0) {
+        return status::StatusOK;
+    }
+
+    return status::StatusErrFile;
+}
```

Comment:
1. `output_` may be NULL, when sink is paused. I think flush should be no-op in this case. (BTW we need to address this in write() too..)
2. `output_->fp` can't be used if SoxSink is device. We should use it only if `driver_type_ == DriverType_File`, otherwise flush should be no-op.
3. Let's also check that `output_->fp` is non-NULL. If it's null, let's make flush a no-op too.
4. nit: Usually early returns are used for errors and the very last return for success.

Reply:
Done, also implemented a fix in write() where buffer will write to sox only if `buffer_pos > 0`